tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -168,6 +186,8 @@ def montecarlo_main_loop(
     virt_packet_last_line_interaction_out_id = []
 
     for i in prange(len(output_nus)):
+        with objmode:
+            update_packet_pbar(1)
```

Comment:
Try using `if (i % 1000 == 0)` as a condition.  You might need to also change the argument in `update_packet_pbar` to 1000.  Use the tqdm documentation here and click the links to the source code for the update function:

https://tqdm.github.io/docs/tqdm/